### PR TITLE
Sign in with email login 

### PR DESCRIPTION
**Pull request by nathanstitt**
screen shot 2016-11-09 at 4 35 36 pm

screen shot 2016-11-11 at 11 29 21 am

### CODE REVIEW

File: app/assets/stylesheets/application_layout.scss [hunk 1]
```diff
@@ -103,7 +103,10 @@ body, p, ol, ul, td, a {
 }
 
 #application-footer {
-  clear: both;
+  position: fixed;
+  left: 0;
+  right: 0;
+  bottom: 0;
```
Comment (nathanstitt) [inline app/assets/stylesheets/application_layout.scss]: Note: The mockups didn't show any copyright info or footer.  I left the existing one in place bug locked it to the bottom of the page.

File: app/views/layouts/_attention.html.erb [hunk 2]
```diff
@@ -1,22 +1,12 @@
...
-  <% alert_messages.push error.translate %>
-<% end %>
-
-<% alert_messages.push(flash[:alert]) if flash[:alert] %>
-
-<% notice_messages = [] %>
-<% notice_messages.push(flash[:notice]) if flash[:notice] %>
-
-<% alert_messages = alert_messages.collect{|msg| msg.gsub("contact support", mail_to("info@openstax.org", "contact support")).html_safe} %>
+<% errors = collect_errors %>
```
Comment (nathanstitt) [inline app/views/layouts/_attention.html.erb]: I extracted this into a method in application_helper.  That way the view can call `collect_errors.any?` to see if it should set an error class on inputs.

File: app/views/sessions/authenticate.html.erb [hunk 3]
```diff
@@ -1,74 +1,46 @@
...
-          <div class="form-group">
-            <label for='password'><%= t :".password" %></label>
-            <input type='password' id='password' name='password' class="form-control"/>
-          </div>
+  <% if login_info[:providers].include?('google_oauth2') %>
+    <%= link_to "/auth/google_oauth2", class: 'btn btn-social btn-google',
+                                       id: 'google-login-button' do %>
+      <span class="fa fa-google"></span> <%= t :".sign_in_with_google" %>
+    <% end %>
+  <% end %>
```
Comment (nathanstitt) [inline app/views/sessions/authenticate.html.erb]: Note: I removed twitter.  I **think** that was the plan?  Easy enough to restore if we want to.

File: app/controllers/sessions_controller.rb [hunk 4]
```diff
@@ -41,7 +43,7 @@ def lookup_login
                   set_login_info(username_or_email: @handler_result.outputs.username_or_email,
                                  names: @handler_result.outputs.names,
                                  providers: @handler_result.outputs.providers)
-                  redirect_to :authenticate
+                  render :authenticate
```
Comment (jpslav) [inline app/controllers/sessions_controller.rb]: I was doing `redirect_to` b/c I wanted the URL to change.  Not the best use, I know... but having `lookup_login` in the URL seems odd?

File: app/views/layouts/_attention.html.erb [hunk 5]
```diff
@@ -1,22 +1,12 @@
...
 
 <div class="row">
   <div class="col-xs-12">
     <noscript>
-      <%= attention_tag (t :".please_enable_javascript"), :alert %>
+      <%= attention_tag('please_enable_javascript', :alert) %>
```
Comment (jpslav) [inline app/views/layouts/_attention.html.erb]: did the `t` go away on purpose?

File: app/views/sessions/authenticate.html.erb [hunk 6]
```diff
@@ -1,74 +1,46 @@
...
-      <%= link_to "/auth/google_oauth2", class: 'btn btn-block btn-social btn-google',
-                                         id: 'google-login-button' do %>
-        <span class="fa fa-google"></span> <%= t :".sign_in_with_google" %>
-      <% end %>
-    <% end %>
+      <%= f.password_field :password, placeholder: t(".password"), class: 'wide'  %>
 
-  </div>
-</div>
+      <%= f.hidden_field :username_or_email, placeholder: t(".password"),
```
Comment (jpslav) [inline app/views/sessions/authenticate.html.erb]: remove placeholder

File: app/views/sessions/new.html.erb [hunk 7]
```diff
@@ -1,18 +1,28 @@
-<% @hide_session_info = true %>
+<%= image_tag('openstax-logo-white.png', width: 174, height: 40, class: 'logo') %>
```
Comment (jpslav) [inline app/views/sessions/new.html.erb]: should this image be on other pages too?

File: spec/features/user_signs_in_spec.rb [hunk 8]
```diff
@@ -1,45 +1,45 @@
 require 'rails_helper'
 
-xfeature 'User logs in as a local user', js: true do
+feature 'User logs in as a local user', js: true do
 
   background { load 'db/seeds.rb' }
 
-  scenario 'authenticates against the default (bcrypt) password hashes' do
+  scenario 'authentication on the happy path' do
```
Comment (jpslav) [inline spec/features/user_signs_in_spec.rb]: can this be "authentication using passwords"?

File: spec/features/user_signs_in_spec.rb [hunk 9]
```diff
@@ -189,27 +191,6 @@
 
   end
 
-  scenario 'with an email address and password' do
```
Comment (jpslav) [inline spec/features/user_signs_in_spec.rb]: why did this one go away?

File: app/controllers/sessions_controller.rb (hunk 4 above)
Comment (nathanstitt) [inline app/controllers/sessions_controller.rb]: Yeah, I swapped it to `render` so it'd retain the email value for the next form's hidden field, where it'd eventually be passed onto the `custom_identity` strategy.   However we could also use the signed cookie for that as well.  I've made a change for that.

File: app/views/layouts/_attention.html.erb (hunk 5 above)
Comment (nathanstitt) [inline app/views/layouts/_attention.html.erb]: No not really..  At one point I played setting alerts using translation keys which the alert_tag would call `.t` on.  I later decided that was a bad idea, but looks like I forgot to switch this back.

File: app/views/sessions/authenticate.html.erb (hunk 6 above)
Comment (nathanstitt) [inline app/views/sessions/authenticate.html.erb]: Yep, also other attributes that don't make sense hidden fields, like class and autofocus.  (as you can tell I copied the `text_field` for that 😁

File: app/views/sessions/new.html.erb (hunk 7 above)
Comment (nathanstitt) [inline app/views/sessions/new.html.erb]: Yes, the second screen with password/social choices should also have it.  I've extracted into a partial for use on both screens.

File: spec/features/user_signs_in_spec.rb (hunk 9 above)
Comment (nathanstitt) [inline spec/features/user_signs_in_spec.rb]: it was identical to the "happy path", which i've now  renamed to "authentication using email address and password"

**Comment by nathanstitt** [comment]
Thanks @jpslav, ready for another review.

**Comment by jpslav** [comment]
Pushed a fix but travis is currently broken.  Passes locally so merging.

**Pull request merged**